Names/Names/Program.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+
+namespace Names
+{
+    class Program
+    {
+        static void Main()
+        {
+            var names = new List<string>();
+
+            Console.WriteLine("Обрабатываю данные");
+
+            try
+            {
+                using (StreamReader sr = new StreamReader("names.txt"))
+                {
+                    string line;
+                    while((line = sr.ReadLine())!= null)
+                    {
+                        Console.Write('.');
+                        names.Add(line);
```

Comment:
Алгоритм обработки выбран неудачно Вы сначала загружаете все имена в список names (т. е. в оперативную память), а затем проходите по этому списку и формируете результирующий список. И точки печатаете не при добавлении имени в результирующий список, а при чтении файла . Слишком много точек!
Не лучше ли формирование результирующего списка производит сразу при чтении?
```suggestion
                       if (!names.Contains(line))
                        {
                            Console.Write('.');
                            names.Add(line);
                        }
```